code/scripting/api/objs/hudconfig.cpp

```diff
@@ -0,0 +1,237 @@
+#include "hudconfig.h"
+
+#include "hud/hudconfig.h"
+#include "ui/ui.h"
+#include "scripting/api/objs/color.h"
+#include "iff_defs/iff_defs.h"
+#include "globalincs/alphacolors.h"
+
+namespace scripting {
+namespace api {
+
+gauge_config_h::gauge_config_h() : gauge(-1) {}
+gauge_config_h::gauge_config_h(int l_gauge) : gauge(l_gauge) {}
+
+HC_gauge_region* gauge_config_h::getGauge() const
+{
+	return &HC_gauge_regions[GR_1024][gauge];
+}
+
+int gauge_config_h::getIndex() const
+{
+	return gauge;
+}
+
+const char* gauge_config_h::getName() const
+{
+	return HC_gauge_descriptions(gauge);
+}
+
+hud_preset_h::hud_preset_h() : preset(-1) {}
+hud_preset_h::hud_preset_h(int l_preset) : preset(l_preset) {}
+
+int hud_preset_h::getIndex() const
+{
+	return preset;
+}
+
+SCP_string hud_preset_h::getName() const
+{
+	return HC_preset_filenames[preset];
+}
+
+//**********HANDLE: hud preset
+ADE_OBJ(l_HUD_Preset, hud_preset_h, "hud_preset", "Hud preset handle");
+
+ADE_VIRTVAR(Name, l_HUD_Preset, nullptr, "The name of this preset", "string", "The name")
+{
+	hud_preset_h current;
+	if (!ade_get_args(L, "o", l_HUD_Preset.Get(&current))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", current.getName().c_str());
+}
+
+ADE_FUNC(deletePreset, l_HUD_Preset, nullptr, "Deletes the preset file", nullptr, nullptr)
+{
+	hud_preset_h current;
+	if (!ade_get_args(L, "o", l_HUD_Preset.Get(&current))) {
+		return ADE_RETURN_NIL;
+	}
+
+	hud_config_delete_preset(current.getName());
+
+	return ADE_RETURN_NIL;
+}
+
+//**********HANDLE: gauge config
+ADE_OBJ(l_Gauge_Config, gauge_config_h, "gauge_config", "Gauge config handle");
+
+ADE_VIRTVAR(Name, l_Gauge_Config, nullptr, "The name of this gauge", "string", "The name")
+{
+	gauge_config_h current;
+	if (!ade_get_args(L, "o", l_Gauge_Config.Get(&current))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
+	return ade_set_args(L, "s", current.getName());
+}
+
+ADE_VIRTVAR(CurrentColor,
+	l_Gauge_Config,
+	"color",
+	"Gets the current color of the gauge. If setting the color, gauges that use IFF for color cannot be set.",
+	"color",
+	"The gauge color")
+{
+	gauge_config_h current;
+	color newColor;
+	if (!ade_get_args(L, "o|o", l_Gauge_Config.Get(&current), l_Color.Get(&newColor))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (ADE_SETTING_VAR) {
+		if (!current.getGauge()->use_iff) {
+			HUD_config.clr[current.getIndex()] = newColor;
+		}
+	}
+
+	color thisColor;
```

Comment:
I'd recommend using `const color *`.  It avoids the need to copy the entire `color` struct.